### PR TITLE
Resolved TODO in Settings

### PR DESCRIPTION
**Pull request by Grenkin1988**
Resolved:
`// TODO: Make generic getSettingAs that infers return type from default value.`

### CODE REVIEW

**Comment by vogonistic** [comment]
You have merge conflicts.

**Comment by Grenkin1988** [comment]
@vogonistic Resolved conflicts

File: Assets/Scripts/Utilities/Settings.cs [hunk 1]
```diff
@@ -37,31 +34,34 @@ public static class Settings
...
         {
-            settingsDict.Add(key, defaultValue);
-            SaveSettings();
+            Debug.ULogErrorChannel("Settings", "Settings Dictionary was not loaded!");
             return defaultValue;
         }
-        else
+
+        string setting = GetSetting(key, defaultValue);
+        if (setting != null && setting != defaultValue)
```
Comment (vogonistic) [inline Assets/Scripts/Utilities/Settings.cs]: You should remove the check if `setting != defaultValue`. If it is the same as default value, there is no need to add it and you can safely return the value.

Comment (vogonistic) [inline Assets/Scripts/Utilities/Settings.cs]: As it is, it causes the localization not to load properly for me which generates a whole string of errors.

**Comment by Grenkin1988** [comment]
@vogonistic Fixed

**Pull request merged**